repl/scala-2.11/src/main/scala/org/apache/spark/repl/Main.scala

```diff
@@ -78,16 +79,21 @@ object Main extends Logging {
     }
     sparkContext = new SparkContext(conf)
     logInfo("Created spark context..")
+    sqlContextWithHive = conf.getBoolean("spark.sql.hive.context", true)
     sparkContext
   }
 
   def createSQLContext(): SQLContext = {
-    val name = "org.apache.spark.sql.hive.HiveContext"
+    val name = sqlContextWithHive match {
+      case true => "org.apache.spark.sql.hive.HiveContext"
+      case false => "org.apache.spark.sql.SQLContext"
+    }
+
     val loader = Utils.getContextOrSparkClassLoader
     try {
       sqlContext = loader.loadClass(name).getConstructor(classOf[SparkContext])
         .newInstance(sparkContext).asInstanceOf[SQLContext]
-      logInfo("Created sql context (with Hive support)..")
+      logInfo("Created sql context with " + name)
     } catch {
       case _: java.lang.ClassNotFoundException | _: java.lang.NoClassDefFoundError =>
         sqlContext = new SQLContext(sparkContext)
```

Comment:
I think it makes sense to just abort if we fail to instantiate the context instance. But should provide informative information and suggestion about 1) the newly added option, and 2) if `HiveContext` is required, ask the user to build Spark with `-Phive`.